### ISSUE TITLE
INGV data not showing up through eida-routing

### ISSUE THREAD

**Issue by SusiCustodio** [EIDA/userfeedback#135]
Hi,

I can get data from Italian stations through the INGV client, but not through the eida-routing client, see ObsPy code below. 

It used to work a month ago, but not anymore...

Can you fix it?

Thanks,
Susana

....................................................................................................

from obspy.clients.fdsn import Client, RoutingClient
from obspy import UTCDateTime

# client = Client("INGV")
client = RoutingClient("eida-routing")
                                
t0 = UTCDateTime(2022,1,1)		                                
inventory = client.get_stations(station='CING', starttime=t0, endtime=t0+60)

**Comment by jschaeff** [comment]
Hello, thank you for reporting this and having a look into the obspy debug messages.
INGV deployed a new implementation of ws station las week, after a period of tests. But we could not test the routing system.

@petrrr has been notified.

**Comment by SusiCustodio** [comment]
Hi, thank you both Tobias and Jonathan. I actually meant to have reported the message to EIDA but somehow I reported it to obspy (probably just followed some link that was in front of my eyes at the moment). Many thanks for looking into this and for having notified INGV.

**Comment by jschaeff** [comment]
This is the EIDA user support system, and you did right :smile:

**Comment by megies** [comment]
> Hi, thank you both Tobias and Jonathan. I actually meant to have reported the message to EIDA but somehow I reported it to obspy (probably just followed some link that was in front of my eyes at the moment). Many thanks for looking into this and for having notified INGV.

Big brother is watching [the EIDA/userfeedback repository] :smile: 

And no worries, took no time to find the issue.

**Comment by SusiCustodio** [comment]
Oh, glad to hear that it went to the right place! It would not be unheard of for me to make such confusions… 😅
Many thanks again 🙏

**Comment by ilpint** [comment]
@SusiCustodio you should try this workaround, adding the desired level to the call:

`client.get_stations_bulk(bulk=[('IV', 'CING', '*', '*', '2022-01-01T00:00:00', '2022-01-01T00:01:00')], level="channel")`

**Comment by megies** [comment]
@ilpint I don't think that helps, seems to hit the same problem on the server:

```
In [11]: client.get_stations_bulk(bulk=[('IV', 'CING', '*', '*', '2022-01-01T00:00:00', '2022-01-01T00:01:00')], level="channel")
Downloading http://webservices.ingv.it/fdsnws/station/1/query with requesting gzip compression
Sending along the following payload:
----------------------------------------------------------------------
level=channel
IV CING * * 2022-01-01T00:00:00 2022-01-01T00:01:00
----------------------------------------------------------------------
HTTP error 400, reason Bad Request, while downloading 'http://webservices.ingv.it/fdsnws/station/1/query': b'<?xml version="1.0" ?><exception><path>/db/apps/fdsn-station/modules/query.xql</path><message>exerr:ERROR XPTY0004: The actual cardinality for parameter 1 does not match the cardinality declared in the function\'s signature: stationutil:get-parameter($m as map(*), $k as xs:string) xs:string. Expected cardinality: exactly one, got 0. [at line 1658, column 5, source: /db/apps/fdsn-station/modules/util.xql]\nIn function:\n\tstationutil:get-parameter(map(*), xs:string) [1658:5:/db/apps/fdsn-station/modules/util.xql]\n\tstationutil:request_error(map(*)*) [1849:1:/db/apps/fdsn-station/modules/util.xql]\n\tstationutil:run() [21:71:/db/apps/fdsn-station/modules/util.xql]</message></exception>'
```

**Comment by ilpint** [comment]
@megies, I think you was hitting a cached result. 
Meanwhile I changed a proxy configuration variable and it should be solved. 
It works (for me) also without level specification.

**Comment by megies** [comment]
Ah, yes, seems to work now @ilpint :+1: 
I think this can be closed.

**Comment by SusiCustodio** [comment]
Works for me too. Many thanks!

**Comment by petrrr** [comment]
@ilpint Thanks for have fixed this problem immediately.

There is however a detail which still puzzles me and which we should understand better before closing the issue. 

The configuration on the reverse proxy @ilpint is referring to (at least as far as I recall / understand) was supposed to work around an incorrect behavior of the Obspy client (wrong  `Content-Type` header in the POST request). This problem was already addressed in https://github.com/obspy/obspy/issues/3139, so in theory it should have been resolved in the most recent versions.

Now I (we) do not have the information on the Obspy version @SusiCustodio used. But I used version 1.3.1 to confirm the problem and I somehow suspect that @megies used the most recent 1.4.0. For these recent versions I would have expect that the "hack" on the reverse proxy would no be necessary.

Yes, we still what to support older Obspy versions as well, so the proxy hack is necessary. But we also should understand why the supposed Obspy fix does not work as expected.
